### PR TITLE
Revert "[web:tools] always use CanvasKit from the cache when building web apps (#93002)"

### PR DESCRIPTION
**Pull request by mdebbar**
This reverts commit 773789344e80d4fd5e245bf1b6f802aaa2d3e6b5.

Now that CanvasKit is built from sources, let's revert https://github.com/flutter/flutter/pull/93002 so that the locally built CanvasKit is copied to the app's `build/web/` folder.

Depends on https://github.com/flutter/engine/pull/38448

Part of https://github.com/flutter/flutter/issues/118799

### CODE REVIEW

**Review by eyebrowsoffire — APPROVED** [review]
LGTM!

**Comment by mdebbar** [comment]
@eyebrowsoffire in order for this to fully work, we need to change our engine build so that the canvaskit files are generated in `out/wasm_release/canvaskit/` instead of directly under `out/wasm_release/`, right? (If so, my [other PR](https://github.com/flutter/engine/pull/38448) does that).

**Comment by eyebrowsoffire** [comment]
Sounds good to me. It sounds like @zanderso has some notes on that PR and so we'll wait on merging this until those are resolved.

**Comment by christopherfujino** [comment]
@mdebbar are you still planning to land this?

**Comment by mdebbar** [comment]
cc @yjbanov I would love to get your review on this one since you are the author of the PR being reverted.

**Review by yjbanov — COMMENTED** [review]
I think you missed a spot. Otherwise, LGTM.

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 1]
```diff
@@ -511,15 +510,7 @@ class WebBuiltInAssets extends Target {
 
   @override
   Future<void> build(Environment environment) async {
-    // TODO(yjbanov): https://github.com/flutter/flutter/issues/52588
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:514]: Ah, that TODO was added in a different PR than the one I'm reverting. That's why I didn't catch it.

I also made other changes:

- Use the new `Artifacts.canvasKitPath` to accurately locate the canvaskit files.
- Stop removing the `/flutter_web_sdk/` segment from paths in `bin/cache` (it's not needed anymore after https://github.com/flutter/engine/pull/39072).

@yjbanov PTAL :)

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**